tree/tree/src/TTree.cxx

```diff
@@ -1494,7 +1494,7 @@ Long64_t TTree::AutoSave(Option_t* option)
       nbytes = fDirectory->WriteTObject(this,"","overwrite");
    } else {
       nbytes = fDirectory->WriteTObject(this); //nbytes will be 0 if Write failed (disk space exceeded)
-      if (nbytes && key) {
+      if (nbytes && key && strcmp("TTree", key->GetClassName()) == 0) {
```

Comment:
Could this be a derived class name, e.g. `TNTuple`?

Reply:
Good point !